salt/pillar/netbox.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+'''
+A module that adds data to the Pillar structure from a NetBox API.
+
+
+Configuring the NetBox ext_pillar
+====================================
+
+.. code-block:: yaml
+
+  ext_pillar:
+    - netbox:
```

Comment:
I would argue that this Pillar module should not be named as `netbox`, as its scope is very, very specific to retrieving data from NetBox and populating the `proxy` key into the Pillar. While this is very useful and nice, maybe we should emphasise the scope and rename it, say `netbox_proxy` or something similar.

To clarify: a `netbox` external Pillar I would expect to provide me _all_ the NetBox data related to that device (i.e, from the  `dcim/devices/<device ID>/` endpoint).

Reply:
It's creating the `proxy` key if requested, but this isn't just doing that. It also is returning the device info in the `netbox` key. It's a raw dump of what it gets back from the API call.

Reply:
Ah, okay, I misread that - please disregard.